src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -744,3 +745,157 @@ def _fetch_class_library_tuple(module):
+        config,
+        load_connected_pipeline=load_connected_pipeline,
+        custom_pipeline=custom_pipeline,
+        class_name=custom_class_name,
+        cache_dir=cache_dir,
+        revision=custom_revision,
+    )
+    return custom_pipeline, pipeline_class
+
+
+def _maybe_raise_warning_for_inpainting(pipeline_class, pretrained_model_name_or_path: str, config: dict):
+    if pipeline_class.__name__ == "StableDiffusionInpaintPipeline" and version.parse(
+        version.parse(config["_diffusers_version"]).base_version
+    ) <= version.parse("0.5.1"):
+        from diffusers import StableDiffusionInpaintPipeline, StableDiffusionInpaintPipelineLegacy
+
+        pipeline_class = StableDiffusionInpaintPipelineLegacy
+
+        deprecation_message = (
+            "You are using a legacy checkpoint for inpainting with Stable Diffusion, therefore we are loading the"
+            f" {StableDiffusionInpaintPipelineLegacy} class instead of {StableDiffusionInpaintPipeline}. For"
+            " better inpainting results, we strongly suggest using Stable Diffusion's official inpainting"
+            " checkpoint: https://huggingface.co/runwayml/stable-diffusion-inpainting instead or adapting your"
+            f" checkpoint {pretrained_model_name_or_path} to the format of"
+            " https://huggingface.co/runwayml/stable-diffusion-inpainting. Note that we do not actively maintain"
+            " the {StableDiffusionInpaintPipelineLegacy} class and will likely remove it in version 1.0.0."
+        )
+        deprecate("StableDiffusionInpaintPipelineLegacy", "1.0.0", deprecation_message, standard_warn=False)
+
+
+def _filter_null_components(init_dict: dict, passed_class_objs: dict) -> dict:
+    def load_module(name, value):
+        if value[0] is None:
+            return False
+        if name in passed_class_objs and passed_class_objs[name] is None:
+            return False
+        return True
+
+    init_dict = {k: v for k, v in init_dict.items() if load_module(k, v)}
+    return init_dict
+
+
+def _determine_current_device_map(device_map: dict, component_name: str) -> Tuple[None, dict]:
+    current_device_map = None
+    if device_map is not None and len(device_map) > 0:
+        component_device = device_map.get(component_name, None)
+        if component_device is not None:
+            current_device_map = {"": component_device}
+    return current_device_map
+
+
+def _update_init_kwargs_with_connected_pipeline(
+    init_kwargs: dict, passed_pipe_kwargs: dict, passed_class_objs: dict, folder: str, **pipeline_loading_kwargs
+) -> dict:
+    from .pipeline_utils import DiffusionPipeline
+
+    modelcard = ModelCard.load(os.path.join(folder, "README.md"))
+    connected_pipes = {prefix: getattr(modelcard.data, prefix, [None])[0] for prefix in CONNECTED_PIPES_KEYS}
+    load_kwargs = {
+        "cache_dir": pipeline_loading_kwargs.get("cache_dir"),
+        "force_download": pipeline_loading_kwargs.get("force_download"),
```

Comment:
so for example `force_download` is default to `False` and same as `from_flax` etc https://github.com/huggingface/diffusers/blob/16a3dad474dad00f8e4071d699e1562471a2dacd/src/diffusers/pipelines/pipeline_utils.py#L626

and custom_pipeline is modified here https://github.com/huggingface/diffusers/blob/16a3dad474dad00f8e4071d699e1562471a2dacd/src/diffusers/pipelines/pipeline_utils.py#L742

so don't think pass the `kwargs.copy()` is the approach here

Reply:
in that case, we do not need to do anything, just pass `**pipeline_loading_kwargs`, no? cc @DN6 here too

Reply:
Yeah exactly. Done in 7e811bfac.

Reply:
But do also note this: 5519f8280